src/Interpreters/ExpressionAnalyzer.h

```diff
@@ -331,7 +332,7 @@ class SelectQueryExpressionAnalyzer : public ExpressionAnalyzer
     SelectQueryOptions query_options;
 
     /// Create Set-s that we make from IN section to use index on them.
-    void makeSetsForIndex(const ASTPtr & node);
+    void makeSetsForIndex(const ASTPtr & node, bool is_explain_);
```

Comment:
`is_explain` is not a very good name because it is not clear what this flag changes.
Maybe it should be like `only_analize` or `do_not_fill_sets` or something.

Reply:
Also, if we don't create sets for index, `explain indexes` will show wrong result (we need this set to make PK work).
I think we still should create indexes in this case.

Probably, we also need setting for explain which will define if we should create indexes.